### ISSUE TITLE
thread '<unnamed>' panicked at 'no entry found for key'

### ISSUE THREAD

**Issue by pnorman** [pka/t-rex#48]
I have this layer definition

```toml
[[tileset.layer]]
name = "transport"
geometry_field = "way"
geometry_type = "LINESTRING"
srid = 3857
buffer-size = 128
simplify = true
  [[tileset.layer.query]]
  minzoom = 6
  maxzoom = 10
  sql = """
  SELECT
      way,
      name,
      ref,
      highway AS class,
      CASE
        WHEN bridge IS NOT NULL AND bridge <> 'no' THEN 'bridge'
        ELSE CASE WHEN tunnel IS NOT NULL and tunnel <> 'no' THEN 'tunnel' END
      END AS brunnel
    FROM planet_osm_roads
    WHERE way && !bbox!
      AND (highway = 'motorway'
        OR highway = 'trunk' AND !zoom! >= 7
        OR highway = 'primary' AND !zoom! >= 8
        OR highway = 'secondary' AND !zoom! >= 9)
  """
```

and get this error when I try the X-Ray view

```
2017-07-13 05:16:51.608 INFO GET /vector_demo/1/0/1.pbf
thread '<unnamed>' panicked at 'no entry found for key', /checkout/src/libcore/option.rs:794
note: Run with `RUST_BACKTRACE=1` for a backtrace.
```

Turning on the backtrace gives

```
thread '<unnamed>' panicked at 'no entry found for key', /checkout/src/libcore/option.rs:794
stack backtrace:
   0: std::sys::imp::backtrace::tracing::imp::unwind_backtrace
             at /checkout/src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::panicking::default_hook::{{closure}}
             at /checkout/src/libstd/sys_common/backtrace.rs:71
             at /checkout/src/libstd/sys_common/backtrace.rs:60
             at /checkout/src/libstd/panicking.rs:355
   2: std::panicking::rust_panic_with_hook
             at /checkout/src/libstd/panicking.rs:371
             at /checkout/src/libstd/panicking.rs:549
   3: std::panicking::begin_panic
             at /checkout/src/libstd/panicking.rs:511
   4: std::panicking::begin_panic_fmt
             at /checkout/src/libstd/panicking.rs:495
   5: core::panicking::panic_fmt
             at /checkout/src/libstd/panicking.rs:471
   6: core::option::expect_failed
             at /checkout/src/libcore/option.rs:794
   7: <t_rex::datasource::postgis::PostgisInput as t_rex::datasource::datasource::DatasourceInput>::retrieve_features
   8: t_rex::service::mvt::MvtService::tile
   9: <T as nickel::middleware::Middleware<D>>::invoke
  10: <nickel::router::router::Router<D> as nickel::middleware::Middleware<D>>::invoke
  11: <nickel::server::ArcServer<D> as hyper::server::Handler>::handle
  12: hyper::server::listener::spawn_with::{{closure}}
  13: <F as alloc::boxed::FnBox<A>>::call_box
  14: std::sys::imp::thread::Thread::new::thread_start
             at /checkout/src/liballoc/boxed.rs:650
             at /checkout/src/libstd/sys_common/thread.rs:21
             at /checkout/src/libstd/sys/unix/thread.rs:84
  15: start_thread
  16: __clone
```

**Comment by pnorman** [comment]
Increasing the minzoom of the query makes the error go away, but it's not really a fix since I can't have the layer start at z0, and the error message is cryptic.

**Comment by pka** [comment]
Fixed in c721057 - Thanks for reporting!
